### PR TITLE
feature: unify channel navigator in web and desktop app

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
this PR aims at unifying the channel navigator between web and desktop app when the global header is enabled.
there is also a small fix included that aligns the header info burger button.

#### Ticket Link
[MM-37834](https://mattermost.atlassian.net/browse/MM-37834)

#### Release notes
```release-note
test
```

### CODE REVIEW

**Comment by matthewbirtch** [comment]
@michelengelen the test server failed

**Comment by michelengelen** [comment]
@matthewbirtch probably due to the failing tests .... will fix them and then start a new spinwick

**Review by matthewbirtch — APPROVED** [review]
Verified that the desktop app layout for the channel navigator now is the same as the web app.
image

**Review by jgilliam17 — APPROVED** [review]
Thanks @michelengelen 
Tested, looks good to merge.
- Verified, with global header enabled, channel navigator layout on desktop matches the webapp.

File: components/sidebar/sidebar_header/dropdown/sidebar_header_dropdown_button.tsx [hunk 1]
```diff
@@ -111,9 +104,9 @@ export default class SidebarHeaderDropdownButton extends React.PureComponent<Pro
                     id='headerInfo'
                     className='header__info'
                 >
-                    <VerticalStack>
-                        {teamNameWithToolTip}
-                        {!this.props.globalHeaderEnabled &&
+                    {this.props.globalHeaderEnabled ? teamNameWithToolTip : (
```
Comment (deanwhillier) [inline components/sidebar/sidebar_header/dropdown/sidebar_header_dropdown_button.tsx:107]: Apologies Michel, I missed this in my first quick scan. The channel header has been addressed in the [Channel Header PR](https://github.com/mattermost/mattermost-webapp/pull/8569) as well.

**Review by deanwhillier — CHANGES_REQUESTED** [review]
LGTM apart from the one duplicated effort ... again apologies for not catching this earlier.

File: components/sidebar/sidebar_header/dropdown/sidebar_header_dropdown_button.tsx (hunk 1 above)
Comment (michelengelen) [reply components/sidebar/sidebar_header/dropdown/sidebar_header_dropdown_button.tsx]: I just now reverted the changes

**Review by nevyangelova — APPROVED** [review]
lgtm

**Comment by michelengelen** [comment]
@deanwhillier please re-review and merge when everything is fine.

**Review by deanwhillier — APPROVED** [review]

**Pull request merged**